snapcraft/internal/lifecycle.py

```diff
@@ -243,6 +245,18 @@ def _run_step(self, step, part, part_names):
         common.env.extend(self.config.project_env())
 
         part = _replace_in_part(part)
+
+        # Add the annotated list of build packages
+        part_build_packages = part._part_properties['build-packages']
+        part.build_packages = []
+        for pkg in self.build_packages:
+            if pkg in part_build_packages:
+                part.build_packages.append(pkg)
+            else:
+                pkg_name, version = repo._deb._get_pkg_name_parts(pkg)
```

Comment:
This is calling a private method on a private module. Note also that the build-packages syntax is package-format-agnostic, so there's no reason to keep this function specific to debs. Can we promote this method within repo to somewhere more public? `repo.<something>` would be nice.